dev-client/src/screens/HomeScreen/components/SiteListBottomSheet.tsx

```diff
@@ -30,82 +29,109 @@ import {useGeospatialContext} from 'terraso-mobile-client/context/GeospatialCont
-    const backgroundStyle = useMemo(
-      () => ({backgroundColor: colors.grey[300]}),
-      [colors],
-    );
-    const {siteDistances} = useGeospatialContext();
+      const useDistance = useMemo(
+        () => siteDistances !== null,
+        [siteDistances],
+      );
 
-    const useDistance = useMemo(() => siteDistances !== null, [siteDistances]);
+      const ListEmptyComponent = useMemo(
+        () => <Text>{t('site.search.no_matches')}</Text>,
+        [t],
+      );
 
-    return (
-      <BottomSheet
-        ref={ref}
-        snapPoints={snapPoints}
-        index={snapIndex}
-        backgroundStyle={backgroundStyle}
-        handleIndicatorStyle={{backgroundColor: colors.grey[800]}}>
-        <Column px="16px">
-          <Row justifyContent="space-between" alignItems="center" pb="4">
-            <Heading variant="h6">{t('site.list_title')}</Heading>
-          </Row>
-          {sites.length >= 0 && <SiteFilterModal useDistance={useDistance} />}
-        </Column>
-        {isLoadingData ? (
-          <Spinner size="lg" />
-        ) : sites.length === 0 ? (
-          <EmptySiteMessage />
-        ) : (
-          <BottomSheetFlatList
-            style={listStyle}
-            data={filteredSites}
-            keyExtractor={site => site.id}
-            renderItem={renderSite}
-            ItemSeparatorComponent={() => <Box h="8px" />}
-            ListFooterComponent={<Box h="10px" />}
-            ListEmptyComponent={<Text>{t('site.search.no_matches')}</Text>}
-          />
-        )}
-      </BottomSheet>
-    );
-  },
+      return (
+        <BottomSheet
+          ref={ref}
+          snapPoints={snapPoints}
+          index={snapIndex}
+          backgroundStyle={backgroundStyle}
+          handleIndicatorStyle={{backgroundColor: colors.grey[800]}}>
+          <Column px="16px">
+            <Row justifyContent="space-between" alignItems="center" pb="4">
+              <Heading variant="h6">{t('site.list_title')}</Heading>
+            </Row>
+            {sites.length >= 0 && <SiteFilterModal useDistance={useDistance} />}
+          </Column>
+          {isLoadingData ? (
+            <Spinner size="lg" />
+          ) : sites.length === 0 ? (
+            <EmptySiteMessage />
+          ) : (
+            <BottomSheetFlatList
+              style={listStyle}
+              data={filteredSites}
+              maxToRenderPerBatch={MAX_TO_RENDER_PER_BATCH}
+              windowSize={WINDOW_SIZE}
+              keyExtractor={keyExtractor}
+              renderItem={renderSite}
+              ItemSeparatorComponent={ItemSeparatorComponent}
+              ListFooterComponent={ListFooterComponent}
+              ListEmptyComponent={ListEmptyComponent}
+            />
+          )}
+        </BottomSheet>
+      );
+    },
+  ),
 );
+
+const keyExtractor = (site: Site) => site.id;
+const ItemSeparatorComponent = () => <Box h={`${SEPARATOR_HEIGHT}px`} />;
```

Comment:
question: can we use an integer value without `px`?

Reply:
when we have more time to migrate away from NativeBase's API, but right now that would require making changes to the actual component's code in basically every file